Qa/psalm

superrosko opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

 - Create a psalm.xml.dist in the project root
 - Copy and paste the contents from this psalm.xml.dist and fix projectFiles
 - Run $ composer --dev require vimeo/psalm
 - Run $ composer require --dev psalm/plugin-phpunit
 - Run $ vendor/bin/psalm --config=psalm.xml.dist --set-baseline=psalm-baseline.xml
 - Add a composer script static-analysis with the command psalm --shepherd --stats
 - Add a new line to script: in .travis.yml: - if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi
 - Remove phpstan from the project (phpstan.neon.dist, .travis.yml entry, composer.json require-dev and scripts)

Closes #4

superrosko committed:
feat: add psalm

closes #4

Signed-off-by: Dmitriy Bespalov <dbespalov.rosko@gmail.com>

boesing commented:
@geerteltink did you merge-up `laminas/automatic-releases`? Looks like 1.4.x does not contain these changes?

boesing commented:
@Ocramius do we want to use plugins for psalm as in this PR?

boesing reviewed (changes requested):
Waiting for feedback regarding the `automatic-releases` integration.
Besides that, LGTM! Well done @superrosko.
Will mark this PR with the proper labels later on, there are some additional steps to do before this counts in for the hacktoberfest event tho.

superrosko committed:
fix: rename psalm.xml

closes #4

Signed-off-by: Dmitriy Bespalov <dbespalov.rosko@gmail.com>

weierophinney committed:
qa: update baseline, based on newer Psalm version present

Flags more "mixed" argument/return values than previously.

Additionally, added the `MixedReturnType` fixer, which added return annotations where missing.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: fix CS issues

- Fixes issues flagged by CS
- Updates data providers and test methods using data providers to use proper types

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
